tests/area/test_markets.py

```diff
@@ -81,8 +81,10 @@ def test_market_rotation_is_successful_keep_past_markets(self, area_fixture):
         ticks_per_slot = area_fixture.config.slot_length / area_fixture.config.tick_length
         area_fixture.current_tick = ticks_per_slot
         area_fixture.cycle_markets()
+        # The expected future market count is market slots for the whole simulation duration
+        # -2 to not take into account the first 2 market slots due to the cycle_markets call
```

Comment:
Sorry, I do not get that. What changed that 
```python
expected_number_of_future_markets = (
                GlobalConfig.future_market_duration / area_fixture.config.slot_length + 1)
```
is not True any more? The number of future markets is not dependant on the sim duration AFAIU.

Reply:
Doch, I was also surprised at first however it is true. Think of a simulation with 24 hours duration, with future market horizon of 24 hours. In this case, if we do not limit the future markets in the simulation duration, we would end up with future markets being created up until 2 days from the start of the simulation (because future markets are created on every market slot, at the 23:45 we would create a future market for the next day at 23:45). However this does not make sense, because as long as a user sets the simulation duration to 1 day, he only cares about what happens during this day and not after that, even if future markets perform trades on the next day. 
The -2 is simple to deduce, it is because the first future market (00:00 does not exist for the future markets, so 00:15) has already been recycled by the cycle_market call.

Reply:
Ahhh. yes this is what you meant in a personal chat today. You are referring to this change: right?
https://github.com/gridsingularity/gsy-e/pull/1277/files#diff-9f03ce1c1dbc11b431a78472cf3160e5beea1f89caa9ba8e6ce5b5858b45ad7eR143

Can you please push your gsy-framework branch so that i can see the function `is_time_slot_in_simulation_duration` there too? Thanks!

Reply:
This is not framework actually, it is in gsy_e.gsy_e_core.util. It is only a range comparison between start/end date + the CN case.

Reply:
Ah darn, indeed.

Reply:
I leave this comment unresolved for others to read along. But for me it is resolved. :-)